core/src/message_queue.c

```diff
@@ -24,15 +24,20 @@ typedef struct MESSAGE_QUEUE_TAG
 static MESSAGE_HANDLE message_pop(MESSAGE_QUEUE_HANDLE_DATA* handle)
 {
     MESSAGE_HANDLE result;
-    MESSAGE_QUEUE_STORAGE* entry = 
+	if (DList_IsListEmpty((PDLIST_ENTRY)&(handle->queue_head)))
```

Comment:
This function is internal, so the queue gets a null check before use in \_destroy and \_pop.